common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/FormProcessorCreator.java

```diff
@@ -69,6 +71,44 @@ public FormProcessorCreator() {
   @Override
   public ParamValueProcessor create(Parameter parameter, Type genericParamType) {
     JavaType targetType = TypeFactory.defaultInstance().constructType(genericParamType);
+
+    if (isPart(parameter)) {
+      return new PartProcessor(parameter.getName(), targetType);
+    }
     return new FormProcessor(parameter.getName(), targetType);
   }
+
+  private boolean isPart(Parameter parameter) {
+    return new FileProperty().getType().equals(((FormParameter) parameter).getType());
+  }
+
+  private static class PartProcessor extends AbstractParamProcessor {
+    PartProcessor(String paramPath, JavaType targetType) {
+      super(paramPath, targetType);
+    }
+
+    @Override
+    public Object getValue(HttpServletRequest request) throws Exception {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> forms = (Map<String, Object>) request.getAttribute(RestConst.FORM_PARAMETERS);
```

Comment:
file parameter only need request.getPart?

Reply:
part is not created on the consumer side. the consumer only puts files in http entity, which ends up in the form